src/main/java/org/apereo/openlrs/OpenLRSAuthenticationFilter.java

```diff
@@ -111,11 +111,24 @@ private void authenticateOAuth(String authorizationHeader, HttpServletRequest re
 						normalizedParams.put(key, value);
 					}
 				}
-				
+
+				StringBuffer requestURL = request.getRequestURL();
+				String requestScheme = request.getScheme();
+				String originalScheme = request.getHeader("x-forwarded-proto");
+
+				// If original request scheme (maybe through load-balancer) is different, use it instead.
```

Comment:
I found that org.springframework.web.util.UriComponentsBuilder can make a URL from a request object and automatically apply the "x-forwarded-*" headers.  But it doesn't come without drawbacks.